### PR TITLE
Namespace ACL Resource Implementation

### PR DESCRIPTION
**Pull request by RayLiu7**
# Description
Namespace ACL Resource Implementation.

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE NAME
Namespace ACL

##### OUTPUT
image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by RayLiu7** [comment]
run e2e test

File: powerscale/provider/namespace_acl_resource.go [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error creating namespace acl",
+			fmt.Sprintf("Namespace acl param invalid with error: %s", message),
+		)
+		return
+	}
+
+	if namespaceACLToCreate.HasAcl() {
+		err = helper.UpdateNamespaceACL(ctx, r.client, plan, namespaceACLToCreate)
```
Comment (taohe1012) [inline powerscale/provider/namespace_acl_resource.go:340]: better to add cases in test - leave acl_custom  empty or setting acl_custom = []

Comment (taohe1012) [reply powerscale/provider/namespace_acl_resource.go:340]: And seems the resource is similar to settings, do you support `do import action when creating with empty config` and if support, better also to add one test case for it.

File: powerscale/provider/namespace_acl_resource.go [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+}
+
+// NamespaceACLResource defines the resource implementation.
+type NamespaceACLResource struct {
+	client *client.Client
+}
+
+// Metadata describes the resource arguments.
+func (r *NamespaceACLResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_namespaceacl"
```
[the remark this answers is not in the document]
Comment (RayLiu7) [reply powerscale/provider/namespace_acl_resource.go]: Yes, name modified

File: powerscale/provider/namespace_acl_resource.go (hunk 1 above)
Comment (RayLiu7) [reply powerscale/provider/namespace_acl_resource.go:340]: Yes, cases added

**Comment by RayLiu7** [comment]
run e2e test

**Review by P-Cao — APPROVED** [review]

**Comment by RayLiu7** [comment]
run e2e test

**Review by sapana05 — APPROVED** [review]

**Review by MangirishK — APPROVED** [review]

File: powerscale/provider/namespace_acl_resource.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (taohe1012) [reply powerscale/provider/namespace_acl_resource.go]: thanks @RayLiu7

**Review by taohe1012 — APPROVED** [review]
